UI/MapUI.gd

```diff
@@ -0,0 +1,64 @@
+extends Control
+
+var map_scene: PackedScene = preload("res://#Scenes/CardScrollUI.tscn")
+var room_ui: PackedScene = load("res://Map/RoomUI.tscn")
+var _padding_offset = 20
+
+func _input(_inputevent: InputEvent) -> void:
+	if Input.is_key_pressed(KEY_ESCAPE):
+		queue_free()
+
+
+func _on_return_button_press():
+	queue_free()
+
+
+func _ready():
+	var current_map = MapManager.current_map
+	
+	# Create New Room Object to append to the room container
+	var new_room = room_ui.instantiate()
+	var new_room_size = new_room.get_child(0).get_size()
```

Comment:
Could do a function in `RoomUI` script called `get_room_size()` that does this. Might make this section a bit clearer because it would remove the `get_child(0)` call

Reply:
Or alternatively use `Helpers.get_child_node_of_type()` like I mentioned below